Add docker-compose support (#6890)
magento/devdocs: README.md

>rake preview:all
>```

> ## Docker (docker-compose)

Reviewer: To add some context to this feature, can you add the following information in a summary before the instructions:

1. What are the use cases for using this feature?
1. Do users need anything else besides docker installed?
1. Will this work from a fresh copy of the repo or do people need to perform some pre-requisite steps?

Author: It's quite simple. Only a docker installation is needed. Should work in every repo/fork. I can add an additional comment.

Author: @jcalcaben I added an description. Please review it. I am not a native speaker.

>rake preview:all
>```

## Docker (docker-compose)

> Docker provide an easy and fastr way to build the documentation for developers. There is no need to install any

Reviewer: ```suggestion
Docker provides an easy and faster way to build the documentation for developers. There is no need to install any
```

ruby dependency manually. The provided setup starts Jekyll inside the container in watch mode.

> Every change of a markdown file will automatically trigger a rebuild of the documentation.

Author: Do you have the livereload plugin in your browser enabled?

> Every change of a markdown file will automatically trigger a rebuild of the documentation.

Author: Console should print:

`jekyll_1  |         LiveReload: Browser connected`

> Every change of a markdown file will automatically trigger a rebuild of the documentation.

Reviewer 2: Yes, I use it all the time.

> Every change of a markdown file will automatically trigger a rebuild of the documentation.

Reviewer 2: Ah, it's not binding to localhost. I'll figure it out. I might have something else messing it up. 😄

1. Install docker

1. Run docker-compose command:

```bash
docker-compose up
```

> Open your browser (http://127.0.0.1:4000)

Reviewer 2: ```suggestion
1. Use the server address URL provided by the system response.

   For example:
   
   ```terminal
   jekyll_1  |     Server address: http://127.0.0.1:4000/
   jekyll_1  |   Server running... press ctrl-c to stop.
   ```
```

>rake preview:all
>```

## Docker (docker-compose)

> Docker provides a quick and easy way to build and preview the documentation. You do not need to install Ruby dependencies manually. This Docker solution launches Jekyll inside the container in `watch` mode.```

Reviewer 2: Not sure where this came from. Silly GitHub.

```suggestion
Docker provides a quick and easy way to build and preview the documentation. You do not need to install Ruby dependencies manually. This Docker solution launches Jekyll inside the container in `watch` mode.
```

> ruby dependency manually. The provided setup starts Jekyll inside the container in watch mode.

Reviewer 2: You don't need this line now, since the updates were made. 🙂

Every change of a markdown file will automatically trigger a rebuild of the documentation.

1. Install [Docker](https://www.docker.com/get-started). If you have Docker installed, make sure Docker is running.

1. Build and start the Docker environment.

```bash
docker-compose up
```

1. Use the server address URL provided by the system response.
   For example:
   
   ```terminal
   jekyll_1  |     Server address: http://127.0.0.1:4000/
   jekyll_1  |   Server running... press ctrl-c to stop.

> 

Reviewer 2: Missing the closing code block. I'm sure that was a GitHub thing too.

Author: Yes, already fixed :-)
